services/login_service/app/auth/routes.py

```diff
@@ -0,0 +1,151 @@
+from email_verification.email_services import send_verification_email
+from verify_table.store_verification_code import store_verification_code
+
+# from verify_table.delet_row_verify import delete_verification_code
+from datetime import datetime, timedelta, timezone
+from google.cloud import bigquery
+
+app = FastAPI()
+
+auth_router = APIRouter()
+
+
+class SignupData(BaseModel):
+    email: str
+    password: str
+
+
+class LoginData(BaseModel):
+    email: str
+    password: str
+
+
+class VerifyEmailData(BaseModel):
+    email: str
+    verification_code: str
+
+
+@auth_router.get("/protected")
+async def protected_route(user_email: str = Depends(jwt_required)):
+    return {"message": "You have accessed a protected route!", "user_email": user_email}
+
+
+@auth_router.post("/signup")
+async def signup(data: SignupData):
+    email = data.email
+    password = data.password
+
+    if not is_valid_email(email):
+        raise HTTPException(status_code=400, detail="Invalid email format")
+
+    if not is_valid_password(password):
+        raise HTTPException(
+            status_code=400,
+            detail="Password must be at least 8 characters long and contain letters and at least one number",
+        )
+
+    if get_user_from_bigquery(email):
+        raise HTTPException(status_code=400, detail="User already exists")
+
+    hashed_password = generate_password_hash(password)
+
+    email_sent, code = send_verification_email(email)
+    store_verification_code(
+        email=email, verification_code=code, hashed_password=hashed_password
+    )
+
+    if not email_sent:
+        raise HTTPException(status_code=500, detail="Failed to send verification email")
+
+    token = create_access_token({"sub": email})
+    return {
+        "message": f"User {email} created successfully. Please verify your email.",
+        "access_token": token,
+    }
+
+
+@auth_router.post("/login")
+async def login(data: LoginData):
+    email = data.email
+    password = data.password
+
+    user = get_user_from_bigquery(email)
+    if not user:
+        raise HTTPException(status_code=404, detail="User not found")
+
+    if not check_password_hash(user["password"], password):
+        raise HTTPException(status_code=401, detail="Invalid credentials")
+
+    token = create_access_token({"sub": email})
+    return {"access_token": token, "message": "Login successful"}
+
+
+@auth_router.post("/verify_email")
+async def verify_email(data: VerifyEmailData):
+    email = data.email
+    verification_code = data.verification_code
+
+    if not email or not verification_code:
+        raise HTTPException(
+            status_code=400, detail="Email and verification code are required"
+        )
+
+    client = bigquery.Client()
+
+    query = """
+        SELECT verification_code, hashed_password, created_at
+        FROM `annular-net-436607-t0.Instaspy_DS.verification_codes`
+        WHERE email = @email
+        LIMIT 1
+    """
```

Comment:
I prefer to store this kind of queries outside of the main logic part. So you can have file named `queries.py` and store them there, and in code use only meaningful values

```python
EMAIL_VERIFICATION_QUERY = """
        SELECT verification_code, hashed_password, created_at
        FROM `annular-net-436607-t0.Instaspy_DS.verification_codes`
        WHERE email = @email
        LIMIT 1
    """
```